compiler/util-klib-abi/src/org/jetbrains/kotlin/library/abi/parser/KLibDumpParser.kt

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@file:OptIn(ExperimentalLibraryAbiReader::class)
+
+package org.jetbrains.kotlin.library.abi.parser
+
+import org.jetbrains.kotlin.library.BaseKotlinLibrary
+import org.jetbrains.kotlin.library.abi.AbiClass
+import org.jetbrains.kotlin.library.abi.AbiCompoundName
+import org.jetbrains.kotlin.library.abi.AbiDeclaration
+import org.jetbrains.kotlin.library.abi.AbiEnumEntry
+import org.jetbrains.kotlin.library.abi.AbiFunction
+import org.jetbrains.kotlin.library.abi.AbiModality
+import org.jetbrains.kotlin.library.abi.AbiProperty
+import org.jetbrains.kotlin.library.abi.AbiQualifiedName
+import org.jetbrains.kotlin.library.abi.AbiSignatureVersion
+import org.jetbrains.kotlin.library.abi.AbiSignatures
+import org.jetbrains.kotlin.library.abi.AbiType
+import org.jetbrains.kotlin.library.abi.AbiValueParameter
+import org.jetbrains.kotlin.library.abi.AbiValueParameterKind
+import org.jetbrains.kotlin.library.abi.ExperimentalLibraryAbiReader
+import org.jetbrains.kotlin.library.abi.LibraryAbi
+import org.jetbrains.kotlin.library.abi.LibraryManifest
+import org.jetbrains.kotlin.library.abi.LibraryTarget
+import org.jetbrains.kotlin.library.abi.impl.AbiAnnotationListImpl
+import org.jetbrains.kotlin.library.abi.impl.AbiClassImpl
+import org.jetbrains.kotlin.library.abi.impl.AbiConstructorImpl
+import org.jetbrains.kotlin.library.abi.impl.AbiEnumEntryImpl
+import org.jetbrains.kotlin.library.abi.impl.AbiFunctionImpl
+import org.jetbrains.kotlin.library.abi.impl.AbiPropertyImpl
+import org.jetbrains.kotlin.library.abi.impl.AbiSignaturesImpl
+import org.jetbrains.kotlin.library.abi.impl.AbiTopLevelDeclarationsImpl
+import org.jetbrains.kotlin.library.abi.impl.AbiValueParameterImpl
+import java.io.File
+import java.text.ParseException
+
+/**
+ * Parser for klib dump format created by [org.jetbrains.kotlin.library.abi.LibraryAbiRenderer]
+ *
+ * @property klibDump The text of the dump file
+ * @property filePath The file path to add to parse exceptions for clearer debugging (optional)
+ */
+class KlibDumpParser(klibDump: String, private val filePath: String? = null) {
```

Comment:
Could you please annotate the class with `ExperimentalLibraryAbiReader`?
I thought about adding a separate annotation for it, but this is an ABI Reader, so the existing annotation seems to make sense here.